### PR TITLE
add num_classes argument to confusion matrix

### PR DESCRIPTION
**Pull request by c00k1ez**
## What does this PR do?

Add `num_classes` argument to `metrics.ConfusionMatrix`and `functional.confusion_matrix`, update` test_confusion_matrix()` . Discussed this issue with @SkafteNicki in Slack.

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SkafteNicki — APPROVED** [review]
LGTM. Very good job :]

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/metrics/classification.py [hunk 1]
```diff
@@ -107,12 +107,14 @@ class ConfusionMatrix(TensorMetric):
 
     def __init__(
             self,
+            num_classes: Optional[int] = None,
             normalize: bool = False,
             reduce_group: Any = None,
             reduce_op: Any = None,
```
Comment (Borda) [inline pytorch_lightning/metrics/classification.py:113]:
```suggestion
            normalize: bool = False,
            reduce_group: Any = None,
            reduce_op: Any = None,
            num_classes: Optional[int] = None,
```
just for keeping compatibility or is there a strong reason why to have it in front of all arg?

Comment (rohitgr7) [reply pytorch_lightning/metrics/classification.py:113]: it's in front in all other metrics as well.

Comment (c00k1ez) [reply pytorch_lightning/metrics/classification.py:113]: Yeah, I'm just make it like in Accuracy class.

https://github.com/PyTorchLightning/pytorch-lightning/blob/a1ea681c47004599ee5a47a05ddd1b4ea12e60d4/pytorch_lightning/metrics/classification.py#L52-L58

Comment (Borda) [reply pytorch_lightning/metrics/classification.py:113]: yes, ok, hope it does not heart much...
but my point is that if someone is using the func now and pass just argument (not kwargs) it breaks in this new version...

**Pull request merged**